zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/DataSetInfo.scala

```
@@ -41,11 +41,16 @@ object DataSetInfo {
     }
   }
 
+  //Used by: HierarchyField: "levels" -> Json.toJson(hierarchy.levels))) to write from Seq[(String,String)] to JSON
   implicit def tuple2Writes: Writes[Tuple2[String, String]] = Writes[(String, String)](t => Json.obj("level" -> t._1, "field" -> t._2))
 
   def parseLevels(levelSeq: Seq[Map[String, String]]): Seq[(String, String)] = {
-    levelSeq.map {
-      levelMap => (levelMap.get("level").getOrElse(""), levelMap.get("field").getOrElse(""))
+    try {
```

just let it throw :-)